src/Compilers/Core/Portable/Compilation/Compilation.cs

```diff
@@ -70,6 +70,8 @@ public abstract partial class Compilation
         // Protected for access in CSharpCompilation.WithAdditionalFeatures
         protected readonly IReadOnlyDictionary<string, string> _features;
 
+        private Optional<int?> _lazyDataSectionStringLiteralThreshold;
```

Comment:
>Optional<int?> [](http://example.com/codeflow?start=16&length=14)

I am not sure what advantage we are getting from using ```Optional``` structure here. `int?` already has an ability to represent an unknown value.

Reply:
These are different states:
- `Optional.None` = nothing is cached
- `(int?)null` = the value is cached - and it is `null`

Perhaps we could use negative `int` values for some of these but that seems less clear to me.